src/transports/meta/routing.ts

```diff
@@ -1,30 +1,66 @@
 import { EndpointContext } from '../../adapter'
 import { ResponseCache } from '../../cache/response'
 import { AdapterConfig } from '../../config'
-import { AdapterRequest, AdapterResponse, makeLogger } from '../../util'
+import { AdapterRequest, AdapterResponse, makeLogger, RequestGenerics } from '../../util'
 import { AdapterError } from '../../validation/error'
 import { MetaTransport, Transport, TransportDependencies, TransportGenerics } from '..'
+import { InputParameters } from '../../validation'
 
 const logger = makeLogger('RoutingTransport')
 
+export type RoutingTransportGenerics = TransportGenerics & {
+  Request: RequestGenerics & {
+    Params: RequestGenerics['Params'] & {
+      transport: string
```

Comment:
This takes care of `transport` in the data object but I could see someone using it like endpoint and specifying it outside. Should we cover this scenario too?
```
{
   "endpoint": "price",
   "transport": "websocket",
   "data": {...}
}
```

Reply:
No, we discussed options and landed on specifying in the data payload so the cache key would discriminate between transports. I think adding ambiguity will lead to problems, we should reduce the scope as much as possible.

Reply:
Good enough for me 👍